Lklfuse: vfs mount and open flags

ddiss opened the pull request:
```
The following changes since commit b8618cd3c65eec363c0b6052fbf7647952fcba98:

  lklfuse: enable fuse_config.use_ino (2024-06-04 14:32:52 +1000)

are available in the Git repository at:

  https://github.com/ddiss/linux lklfuse_vfs_open_flags

for you to fetch changes up to afa783baffc2255181c6c3af67c53494210eb5fc:

  lklfuse: map more open flags (2024-06-25 22:26:47 +1000)

----------------------------------------------------------------
David Disseldorp (2):
      lklfuse: parse mount flags
      lklfuse: map more open flags

 tools/lkl/lklfuse.c | 146 +++++++++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 144 insertions(+), 2 deletions(-)
```

tavip commented on tools/lkl/lklfuse.c:
```diff
@@ -545,12 +547,119 @@ const struct fuse_operations lklfuse_ops = {
+
+	*remain = '\0';
+	strncpy(tokbuf, opts, sizeof(tokbuf));
+	while ((tok = strtok_r(tb, "\\,", &saveptr)) != NULL) {
```
Do we need \ as a separator here? Do you have some usage examples?

ddiss replied:
Thanks for taking a look at the changes...

> Do we need \ as a separator here?

LKL kernel mount options provided via the `lklfuse -opts=...` parameter are separated with '\,' instead of ',' so that they don't get processed as part of the regular `fuse_opt_parse()` process.
Using `strtok(delim="\\,")` isn't ideal in that we should be handling each two-char '\,' (and '\,') sequence, instead of accepting either as a delimiter, but it does simplify parsing. I'll have a play with using `strstr()` to do things properly.

> Do you have some usage examples?

```
lklfuse -f -o 'fsname=/dev/zram0,subtype=lkl.xfs,type=xfs,opts=rw\,noatime\,filestreams' /dev/zram0 /mnt/test
```

In this case the escaped `opts` parameter value goes through to `lklfuse_parse_vfs_flags()`, where `rw` and `noatime` are consumed as VFS flags (`rw` would invert any prior `ro`); `filestreams` is passed through as a string to `lkl_sys_mount()` and handled as an xfs specific option in `xfs_super.c`.

tavip replied:
Got it now. If you can add a comment block explaining this I am ok with merging this as it is.

ddiss replied:
hmm, your suspicions were correct: although the `opts` escapes are needed, they helpfully get weeded out by libfuse (`process_real_option_group()`) before being processed here, so we don't need to look for `\`.

ddiss commented:
changes since v1:
- add comment explaining opts `\,` and `\=` escapes
- zero-terminate `remaining_mopts` in `lklfuse_parse_vfs_flags()` even if `opts` zero length

tavip reviewed (commented):
LGTM, thanks @ddiss ! We can ignore the x86_64_qemu failure, that test is flaky.
